### ISSUE TITLE
False negative: pylint does not warn of this name-clash

### ISSUE THREAD

**Issue by omarandlorraine** [pylint-dev/pylint#8659]
### Bug description

Here are the contents of `rtc.py`. It's an MCVE, derive from production code used to store and retrieve a datetime on a real-time-clock in some embedded system.

```
#!/usr/bin/python3
# pylint: disable=missing-module-docstring,missing-class-docstring,unused-argument,missing-function-docstring,too-few-public-methods
import datetime

class RtcDriver:
    def __init__(self, i2c_address):
        self._datetime = datetime.datetime.now()

    @property
    def datetime(self) -> datetime.datetime:
        return self._datetime

    @datetime.setter
    def datetime(self, date_time: datetime.datetime) -> None:
        self._datetime = date_time
```

Pylint scores this 10.0/10.0, but there is (at least) one error here. But when I run `./rtc.py`, I get the following error:

```
somebody@somewhere:~/mcve$ ./rtc.py
Traceback (most recent call last):
  File "/home/sam/mcve/./rtc.py", line 5, in <module>
    class RtcDriver:
  File "/home/sam/mcve/./rtc.py", line 15, in RtcDriver
    def datetime(self, date_time: datetime.datetime) -> None:
AttributeError: 'property' object has no attribute 'datetime'
```

I believe this error means that the name of the class's property clashes with the imported datetime module, and therefore I feel like pylint could have warned me that something was wrong. (There are other cases where pylint warns of redefined names, shadowed variables).

### Command used

```shell
pylint ./rtc.py
```

### Pylint output

```shell
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### Expected behavior

Some kind of warning of a name-clash

### Pylint version

```shell
pylint 2.17.3
astroid 2.15.4
Python 3.9.2 (default, Feb 28 2021, 17:03:44)
[GCC 10.2.1 20210110]
```

### OS / Environment

Debian 11

**Comment by jacobtylerwalls** [comment]
Thanks for the clear report. Maybe we could improve inference of properties so that `no-member` is emitted here.
